### DISCUSSION TITLE
Better style `<details>` tags to be differentiated from one another

### DISCUSSION THREAD

**Discussion by triphora** [community/community#32028] [Other Feature Feedback, Questions, & Ideas]
On [my profile readme](https://github.com/wafflecoffee/wafflecoffee/blob/46df795/README.md), I feature several `<details>` tags to avoid cluttering up the page too much. However, I've found that it can become very difficult to differentiate these `<details>` tags from one another, especially when more than one is toggled at once.

<details><summary>Test 1</summary>

Hi!

Here's a lot of text.

Test text, if you would.

</details>
<details><summary>Test 2</summary>

Hello.

This spoiler also contains a considerable amount of example text.

Test text, if you would.

</details>
<details><summary>Test 3</summary>

Salutations.

Contained within this admonition is the most exquisite of information used for experimentation, even greater than what could be accomplished with the greatest _lorem ipsum_.

Test text, if you would.

</details>

I've tried adding `<hr>` tags to the ends of these to show a delimiter between them, but it simply doesn't work well.

<details><summary>Test 1</summary>

Hi!

Here's a lot of text.

Test text, if you would.

<hr></details>
<details><summary>Test 2</summary>

Hello.

This spoiler also contains a considerable amount of example text.

Test text, if you would.

<hr></details>
<details><summary>Test 3</summary>

Salutations.

Contained within this admonition is the most exquisite of information used for experimentation, even greater than what could be accomplished with the greatest _lorem ipsum_.

Test text, if you would.

<hr></details>

It becomes even worse when you nest them!

<details><summary>Nest test</summary>

This text is part of `Nest test`.

<details open=open><summary>Nest test test nest (open by default)</summary>

This text is part of `Nest test test nest`.

</details>

This text is part of `Nest test`. How would you know this is part of `Nest test` and not `Nest test test nest` without me telling you?

</details>

Thus, I would like to suggest styling them so that they are more self-contained from one another, like a spoiler should be. Here are some examples to draw inspiration from:
- [MDN web docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details#try_it): Very simple design but still does the job very well. I've adapted this into a [TamperMonkey script](https://gist.github.com/wafflecoffee/71943f0b2347bf9ba03158fe9c0c4c4b) if anyone wishes to use it.
- [Modrinth](https://staging.modrinth.com/modpack/waffle): Effectively the same as the MDN example, but with a little bit more put into it. [Source CSS is here.](https://github.com/modrinth/knossos/blob/59bb132dd77a65e7a06235add156975d7399fffd/assets/styles/components.scss#L245-L269)
- [Docusaurus](https://docusaurus.io/docs/markdown-features#details): Fancy, pretty blue spoilers with a dropdown animation. Maybe a bit overkill, but still my favorite of the three here.

**Comment by Andre601** [comment]
Some other nice looking examples is PymDownX's details extension alongside the Material for MkDocs theme used: https://facelessuser.github.io/pymdown-extensions/extensions/details/

Example from one of my uses:
![image](https://user-images.githubusercontent.com/11576465/188482910-4f25a723-68db-40b0-8c19-5e75172d6385.png)

**Comment by osfanbuff63** [comment]
In addition, it would be nice if we could get GFM syntax for this. Using the HTML tags are fine but having a way to do it in Markdown would be nice.

**Comment by Marcono1234** [comment]
A workaround for now can be to put the `details` content inside a block quote. But accessibility-wise this might be confusing.

Here is an example
```md
<details>
<summary>Top</summary>

> Some text
>
> <details>
> <summary>First</summary>
>
> > First nested
>
> </details>
>
> <details>
> <summary>Second</summary>
>
> > Second nested
>
> </details>

</details>
```

<details>
<summary>Top</summary>

> Some text
>
> <details>
> <summary>First</summary>
>
> > First nested
>
> </details>
>
> <details>
> <summary>Second</summary>
>
> > Second nested
>
> </details>

</details>

Screenshot:
![Screenshot](https://github.com/community/community/assets/11685886/0020977e-00bc-4c92-a397-37b9b728a920)

(not sure if mixing Markdown and HTML like this could break in the future though; could possibly also directly use `<blockquote>` instead of `>`)
